zenoh-plugin-ros2dds/src/route_service_cli.rs

```diff
@@ -360,7 +360,7 @@ fn route_dds_request_to_zenoh(
     }
 
     let z_bytes: ZBytes = sample.into();
-    let slice = Cow::from(z_bytes);
+    let slice: ZSlice = z_bytes.into();
     let dds_req_buf = slice.as_ref();
 
     let is_little_endian =
```

Comment:
To avoid an expect here, we could write 
```rust
let request_id = match (
    dds_req_buf.get(4..20).map(|x| x.try_into()),
    is_cdr_little_endian(dds_req_buf),
) {
    (Some(Ok(header)), Some(is_little_endian)) => {
        CddsRequestHeader::from_slice(header, is_little_endian)
    }
    (None, Some(x)) => {
        tracing::warn!("Expected CDR Header at DDS_sample buffer[4..20]");
        return;
    },
    (Some(Err(err)), Some(_)) => {
        tracing::warn!("Failed CDR Header to ZSlice: {err} ");
        return;
    }
    (Some(_), None) => {
        tracing::warn!("Check for little endian failed, buf not long enough");
        return;
    }
    (None, None) => {
        tracing::warn!("Expected CDR Header at DDS_sample buffer[4..20] + Check for little endian failed, buf not long enough ");
        return;
    }
};
```

If we implement buffer check here, and in the below buffers, 
we could remove the length check at the beginning.
```rust 
if sample.len() < 20 {
    tracing::warn!("{route_id}: received invalid request: {sample:0x?}");
    return;
}
```

Reply:
Alternatively less verbose, but more to the point: but then we dont opt into specifics of what failed. 
Might be a nice middle ground. 
```rust
let request_id = match (
    dds_req_buf.get(4..20).map(|x| x.try_into()),
    is_cdr_little_endian(dds_req_buf),
) {
    (Some(Ok(header)), Some(is_little_endian)) => {
        CddsRequestHeader::from_slice(header, is_little_endian)
    }
    (_, _) => {
        tracing::warn!("{route_id} : received invalid request: {sample:0x?} -  CddsRequestHeader Could not be created, buffer length <20 ");
        return;
    }
};
```